core/bind/core_bind.cpp

```diff
@@ -1619,12 +1619,17 @@ Error _Directory::open(const String &p_path) {
 		memdelete(d);
 	}
 	d = alt;
+	dir_open = true;
 
 	return OK;
 }
 
+bool _Directory::is_open() const {
+	return dir_open;
+}
+
 Error _Directory::list_dir_begin(bool p_skip_navigational, bool p_skip_hidden) {
-	ERR_FAIL_COND_V_MSG(!d, ERR_UNCONFIGURED, "Directory must be opened before use.");
+	ERR_FAIL_COND_V_MSG(!is_open(), ERR_UNCONFIGURED, "Directory must be opened before use.");
```

Comment:
These checks are meant to prevent a null pointer dereference, which could still happen as far as I know. `d` is indeed initialized in the constructor, but this init could fail, or be removed in the future.

Right now it should be safe enough with `dir_open` (once properly initialized to `false`), but I think it might be safer to check `dir_open && d` in `is_open()`. (Or a single `ERR_FAIL_NULL_V(d, false)` in `is_open()` maybe?